go-controller/pkg/allocator/pod/pod_annotation_test.go

```diff
@@ -233,6 +237,78 @@ func Test_allocatePodAnnotationWithRollback(t *testing.T) {
 			wantReleasedIPsOnRollback: ovntest.MustParseIPNets("192.168.0.3/24"),
 			role:                      types.NetworkRolePrimary,
 		},
+		{
+			name:                   "expect new IP and ipv6 gateway LLA for primary udn layer2 with dual stack",
+			ipam:                   true,
+			idAllocation:           true,
+			persistentIPAllocation: true,
+			args: args{
+				ipAllocator: &ipAllocatorStub{
+					nextIPs: ovntest.MustParseIPNets("192.168.0.3/24", "2010:100:200::3/60"),
+				},
+				idAllocator: &idAllocatorStub{
+					nextID: 100,
+				},
+			},
+			wantUpdatedPod: true,
+			wantPodAnnotation: &util.PodAnnotation{
+				IPs:            ovntest.MustParseIPNets("192.168.0.3/24", "2010:100:200::3/60"),
+				MAC:            util.IPAddrToHWAddr(ovntest.MustParseIPNets("192.168.0.3/24")[0].IP),
+				Gateways:       []net.IP{ovntest.MustParseIP("192.168.0.1").To4(), ovntest.MustParseIP("2010:100:200::1")},
+				GatewayIPv6LLA: util.HWAddrToIPv6LLA(util.IPAddrToHWAddr(ovntest.MustParseIP("100.65.0.4"))),
+				Routes: []util.PodRoute{
+					{
+						Dest: &net.IPNet{
+							IP:   ovntest.MustParseIP("100.65.0.0").To4(),
+							Mask: net.CIDRMask(16, 32),
+						},
+						NextHop: ovntest.MustParseIP("192.168.0.1").To4(),
+					},
+					{
+						Dest:    ovntest.MustParseIPNet("fd99::/64"),
+						NextHop: ovntest.MustParseIP("2010:100:200::1"),
+					},
+				},
+				Role:     types.NetworkRolePrimary,
+				TunnelID: 100,
+			},
+			wantRelasedIDOnRollback:   true,
+			wantReleasedIPsOnRollback: ovntest.MustParseIPNets("192.168.0.3/24", "2010:100:200::3/60"),
+			role:                      types.NetworkRolePrimary,
+		},
+		{
+			name:                   "expect new IP and ipv6 gateway LLA for primary udn layer2 with single stack IPv6",
+			isSingleStackIPv6:      true,
+			ipam:                   true,
+			idAllocation:           true,
+			persistentIPAllocation: true,
+			args: args{
+				ipAllocator: &ipAllocatorStub{
+					nextIPs: ovntest.MustParseIPNets("2010:100:200::3/60"),
+				},
+				idAllocator: &idAllocatorStub{
+					nextID: 100,
+				},
+			},
+			wantUpdatedPod: true,
+			wantPodAnnotation: &util.PodAnnotation{
+				IPs:            ovntest.MustParseIPNets("2010:100:200::3/60"),
+				MAC:            util.IPAddrToHWAddr(ovntest.MustParseIPNets("2010:100:200::3/60")[0].IP),
+				Gateways:       []net.IP{ovntest.MustParseIP("2010:100:200::1")},
+				GatewayIPv6LLA: util.HWAddrToIPv6LLA(util.IPAddrToHWAddr(ovntest.MustParseIP("fd99::4"))),
+				Routes: []util.PodRoute{
+					{
+						Dest:    ovntest.MustParseIPNet("fd99::/64"),
+						NextHop: ovntest.MustParseIP("2010:100:200::1"),
+					},
+				},
+				Role:     types.NetworkRolePrimary,
+				TunnelID: 100,
+			},
+			wantRelasedIDOnRollback:   true,
+			wantReleasedIPsOnRollback: ovntest.MustParseIPNets("2010:100:200::3/60"),
+			role:                      types.NetworkRolePrimary,
+		},
```

Comment:
i assume single single stack v4 not having it is tested by an existing test?

Reply:
it fails earlier, much before our code
at https://github.com/ovn-kubernetes/ovn-kubernetes/blob/6c366bb49d6fa57c80d6b39834952ce1d4d9dbb6/go-controller/pkg/util/node_annotations.go#L700

so i didnt add tests for that, it is either tested already or not reaching the new code (since it fails before that)

Reply:
no I mean I see tests for v6 and dualstack.. so do you mean to say tests for v4 exist already -> a test that checks the code you are adding is not taking effect for v4...

Reply:
I'd say similarly a test for a role:secondary pod to ensure nothing is changing there

Reply:
right added